### PR TITLE
Adding more batch tests, including DEADLINE_EXCEEDED tests

### PR DESCRIPTION
**Pull request by ola-rozenfeld**

### CODE REVIEW

File: go/client/batch_retries_test.go [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+		resp := &repb.BatchUpdateBlobsResponse{
+			Responses: []*repb.BatchUpdateBlobsResponse_Response{
+				// One non-retriable error.
+				{Digest: digest.TestNew("c", 1).ToProto(), Status: &spb.Status{Code: int32(codes.Internal)}},
+				{Digest: digest.TestNew("d", 1).ToProto(), Status: &spb.Status{Code: int32(codes.PermissionDenied)}},
+			},
+		}
+		return resp, nil
+	}
+	// Will not be reached.
```
Comment (ramymedhat) [inline go/client/batch_retries_test.go]: s/Will/Should?

File: go/client/batch_retries_test.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+}
+
+func (f *flakyBatchServer) GetTree(req *repb.GetTreeRequest, stream regrpc.ContentAddressableStorage_GetTreeServer) error {
+	return status.Error(codes.Unimplemented, "")
+}
+
+func (f *flakyBatchServer) BatchUpdateBlobs(ctx context.Context, req *repb.BatchUpdateBlobsRequest) (*repb.BatchUpdateBlobsResponse, error) {
+	f.updateRequests = append(f.updateRequests, req)
+	if f.numErrors < 1 {
+		f.numErrors++
```
Comment (ramymedhat) [inline go/client/batch_retries_test.go]: Can you just increment the number of errors once at the beginning of the function?

File: go/client/batch_retries_test.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+}
+
+type sleepyBatchServer struct {
+	timeout        time.Duration
+	numErrors      int // A counter of DEADLINE_EXCEEDED errors the server has returned thus far.
+	updateRequests int
+	readRequests   int
+}
+
+func (f *sleepyBatchServer) FindMissingBlobs(ctx context.Context, req *repb.FindMissingBlobsRequest) (*repb.FindMissingBlobsResponse, error) {
```
Comment (ramymedhat) [inline go/client/batch_retries_test.go]: nit: s/f/s

File: go/client/batch_retries_test.go (hunk 1 above)
Comment (ola-rozenfeld) [reply go/client/batch_retries_test.go]: I see, you're a pessimist :-)

File: go/client/batch_retries_test.go (hunk 2 above)
Comment (ola-rozenfeld) [reply go/client/batch_retries_test.go]: Yep.

File: go/client/batch_retries_test.go (hunk 3 above)
Comment (ola-rozenfeld) [reply go/client/batch_retries_test.go]: Done.

**Pull request merged**